src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2070,7 +2070,7 @@ private void LogEventArgsMismatches(ParameterInfo[] infos, object?[] args)
                 // Checking to see if the Parameter types (from the Event method) match the supplied argument types.
                 // Fail if one of two things hold : either the argument type is not equal to the parameter type, or the 
                 // argument is null and the parameter type is non-nullable.
-                if ((args[i] != null && (args[i]!.GetType() != pType)) // TODO-NULLABLE: Indexer nullability tracked (https://github.com/dotnet/roslyn/issues/34644)
+                if ((args[i] != null && (args[i]!.GetType() != pType) && !args[i]!.GetType().ImplementInterface(pType)) // TODO-NULLABLE: Indexer nullability tracked (https://github.com/dotnet/roslyn/issues/34644)
```

Comment:
Is there a way to structure the first option so it doesn't call `GetType()` twice, or is that likely to get optimized away?

Reply:
Not without pulling the whole statement out, I think, and doing something like:
```csharp
Type? argType = args[i]?.GetType();
if (argType != null && argType != pType && argType!.ImplementInterface(pType))...
```
I'm inclined to leave it since it shouldn't be adding too much overhead, and is slightly easier to read in my opinion.